packages/common/src/utils/additionalMetrics.ts

```diff
@@ -0,0 +1,41 @@
+import { convertColumnMetric } from '../types/dbt';
+import { CompiledTable } from '../types/explore';
+import { Metric } from '../types/field';
+import { AdditionalMetric } from '../types/metricQuery';
+
+type ConvertAdditionalMetricArgs = {
+    additionalMetric: AdditionalMetric;
+    table: CompiledTable;
+};
+
+export const convertAdditionalMetric = ({
+    additionalMetric,
+    table,
+}: ConvertAdditionalMetricArgs): Metric => {
+    // dimensions with date ends with caps: DAY, MONTH, YEAR
+    // for example `order_date_DAY`, `order_date_MONTH`
+    const baseDimension = additionalMetric.baseDimensionName
+        ? Object.entries(table.dimensions).find(
+              ([name]) =>
+                  name.toLowerCase() === additionalMetric.baseDimensionName,
+          )?.[1]
+        : undefined;
```

Comment:
getting base dimension from an additional metric

Reply:
why do we need to lower case?